src/jit/optimizer.cpp

```diff
@@ -993,8 +993,7 @@ bool Compiler::optIsLoopTestEvalIntoTemp(GenTreePtr testStmt, GenTreePtr* newTes
     // Make sure we have jtrue (vtmp != 0)
     if ((relop->OperGet() == GT_NE) && 
         (opr1->OperGet() == GT_LCL_VAR) &&
-        (opr2->OperGet() == GT_CNS_INT) && 
-        (opr2->gtIntCon.gtIconVal == 0))
+        opr2->IsIntegralConst(0))
```

Comment:
@sivarv left a comment about this before, it's not clear to me if it was addressed. Is allowing longs here valid? I suppose that it should be unless there are problems on x86.